glusterd2/commands/volumes/volume-status.go

```diff
@@ -12,6 +12,41 @@ import (
 	"github.com/gorilla/mux"
 )
 
+func volumeStatusAllHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volumes, e := volume.GetVolumes()
+	if e != nil {
+		logger.WithError(e).Error("Failed to get volumes")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted)
+		return
+	}
+	resp := createVolumeStatusAllResp(volumes)
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+}
+
+func createVolumeStatusAllResp(volumes []*volume.Volinfo) *api.VolumeStatusAllResp {
+	var response api.VolumeStatusAllResp
+
+	for _, volinfo := range volumes {
+		if volinfo.State != volume.VolStarted {
```

Comment:
If a volume isn't started, then we're returning the response. Shouldn't you actually continue instead? This would mean if we're are trying to get status of all the volumes and if any one of the volume is not started, the response will be truncated at that point which seems wrong to me.